File: app/src/main/java/com/example/challenge_consult_cep/feature/search_cep/data/local/entity/CepEntity.java
package com.example.challenge_consult_cep.feature.search_cep.data.local.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;


@Entity(tableName = "cep")
public class CepEntity {

    @PrimaryKey
    @NonNull
    private String code;
    @ColumnInfo
    private String state;
    @ColumnInfo
    private String city;
    @ColumnInfo
    private String neighborhood;
    @ColumnInfo
    private String address;

    public CepEntity(String code, String state, String city, String neighborhood, String address) {
        this.code = code;
        this.state = state;
        this.city = city;
        this.neighborhood = neighborhood;
        this.address = address;
    }

    public String getCode() {
        return code;
    }

    public String getState() {
        return state;
    }

    public String getCity() {
        return city;
    }

    public String getNeighborhood() {
        return neighborhood;
    }

    public String getAddress() {
        return address;
    }
}

File: app/src/main/java/com/example/challenge_consult_cep/feature/search_cep/data/local/entity/CepEntity.kt
package com.example.challenge_consult_cep.feature.search_cep.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey


@Entity(tableName = "cep")
data class CepEntity(
    @PrimaryKey val code: String,
    val state: String,
    val city: String,
    val neighborhood: String,
    val address: String,
)
